### PR TITLE
Add Sections and MO(layer)/TG(layer) Example

### PR DESCRIPTION
**Pull request by thomas-d-11**
## Description

Changes:

1. Added sub-section headings to the portion before the examples.
2. Edited text around sub-headings to better fit with sub-headings.
3. Added a new Example 6, that allows MO(layer) and TG(layer) functionality to be embedded within tap dance functions.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* The FIXME comment at the top of docs/feature_tap_dance.md is addressed, and therefore deleted.

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

My change is a documentation change. I have tested the code from my Example 6 using a MiniVan keyboard (see qmk_firmware/keyboards/thevankeyboards/minivan), and it works as it should. I believe my example code follows the code style of this project, but this is my first commit so I might have missed something.

### CODE REVIEW

**Comment by drashna** [comment]
Just a heads up:
https://github.com/qmk/qmk_firmware/pull/6259

There is some overlap/conflicts there, and I just wanted to make you aware.  It's not merged yet, but it's a better way to handle the custom tapping term for tap dances.

**Comment by thomas-d-11** [comment]
@drashna Thanks for the pointer. I'm very new to GitHub, so please let me know if there are things I can do on my end to help out, and if so any pointers/search terms you could spare would be a huge help.

**Comment by thomas-d-11** [comment]
@drashna I updated docs/feature_tap_dance.md to include your edits and to no longer use `ACTION_TAP_DANCE_FN_ADVANCED_TIME` in my example. I'll keep an eye out for when your work gets merged, and update my example then.

@noroadsleft Let me know if there's anything else I need to do.

Best!

File: docs/feature_tap_dance.md [hunk 1]
```diff
@@ -1,40 +1,54 @@
...
-* `ACTION_TAP_DANCE_FN_ADVANCED_TIME(on_each_tap_fn, on_dance_finished_fn, on_dance_reset_fn, tap_specific_tapping_term)`: This functions identically to the `ACTION_TAP_DANCE_FN_ADVANCED` function, but uses a custom tapping term for it, instead of the predefined `TAPPING_TERM`.
-
-The first option is enough for a lot of cases, that just want dual roles. For example, `ACTION_TAP_DANCE_DOUBLE(KC_SPC, KC_ENT)` will result in `Space` being sent on single-tap, `Enter` otherwise.
+* ~~`ACTION_TAP_DANCE_FN_ADVANCED_TIME(on_each_tap_fn, on_dance_finished_fn, on_dance_reset_fn, tap_specific_tapping_term)`~~: This functions identically to the `ACTION_TAP_DANCE_FN_ADVANCED` function, but uses a custom tapping term for it, instead of the predefined `TAPPING_TERM`.
+    * This is deprecated in favor of the Per Key Tapping Term functionality, as outlined [here](custom_quantum_functions.md#Custom_Tapping_Term). You'd want to check for the specific `TD()` macro that you want to use (such as `TD(TD_ESC_CAPS)`) instead of using this specific.
```
Comment (drashna) [inline docs/feature_tap_dance.md]: Actually, for now, if you could remove the changes from my PR?  This should be fine as it was, for the most part.   And until my PR lands, this won't be accurate.

**Comment by thomas-d-11** [comment]
I finally got to revert those changes. Sorry for the delay.

**Comment by drashna** [comment]
No worries.  We know that not everyone has a lot of time to do stuff. Or it can be easy to get side tracked.  :)

**Pull request merged**